readthedocs/v3/views.py

```diff
@@ -0,0 +1,297 @@
+    metadata_class = SimpleMetadata
+
+
+class ProjectsViewSet(APIv3Settings, NestedViewSetMixin, FlexFieldsMixin,
+                      ListModelMixin, RetrieveModelMixin, GenericViewSet):
+
+    # Markdown docstring is automatically rendered by BrowsableAPIRenderer.
+
+    """
+    Endpoints related to ``Project`` objects.
+
+    * Listing objects.
+    * Detailed object.
+
+    Retrieving only needed data using ``?fields=`` URL attribute is allowed.
+
+    ### Filters
+
+    Allowed via URL attributes:
+
+    * slug
+    * slug__contains
+    * name
+    * name__contains
+
+    ### Expandable fields
+
+    There are some fields that are not returned by default because they are
+    expensive to calculate. Although, they are available for those cases where
+    they are needed.
+
+    Allowed via ``?expand=`` URL attribue:
+
+    * users
+    * active_versions
+    * active_versions.last_build
+    * active_versions.last_build.confg
+
+
+    ### Examples:
+
+    * List my projects: ``/api/v3/projects/``
+    * Filter list: ``/api/v3/projects/?name__contains=test``
+    * Retrieve only needed data: ``/api/v3/projects/?fields=slug,created``
+    * Retrieve specific project: ``/api/v3/projects/{project_slug}/``
+    * Expand required fields: ``/api/v3/projects/{project_slug}/?expand=active_versions``
+    * Translations of a project: ``/api/v3/projects/{project_slug}/translations/``
+    * Subprojects of a project: ``/api/v3/projects/{project_slug}/subprojects/``
+    * Superproject of a project: ``/api/v3/projects/{project_slug}/superproject/``
+    """
+
+    model = Project
+    lookup_field = 'slug'
+    lookup_url_kwarg = 'project_slug'
+    serializer_class = ProjectSerializer
+    filterset_class = ProjectFilter
+    queryset = Project.objects.all()
+    permit_list_expands = [
+        'users',
+        'active_versions',
+        'active_versions.last_build',
+        'active_versions.last_build.config',
+    ]
+
+    # NOTE: accessing a existent project when we don't have permissions to
+    # access it, returns 404 instead of 403.
+
+    def get_view_description(self, *args, **kwargs):
+        """
+        Make valid links for the user's documentation browseable API.
+
+        If the user has already one project, we pick the first and make all the
+        links for that project. Otherwise, we default to the placeholder.
+        """
+        description = super().get_view_description(*args, **kwargs)
+
+        project = None
+        if self.request and self.request.user.is_authenticated():
+            project = self.request.user.projects.first()
+            if project:
+                # TODO: make the links clickable when ``kwargs.html=True``
+                return mark_safe(description.format(project_slug=project.slug))
+        return description
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        return queryset.filter(users=self.request.user)
```

Comment:
Is there an abstraction here that will work better between the .org & .com?

Reply:
Yes. I need to use `.public`/`.api` method form the manager here probably so the migration to .com is easier and just in one place.